Correct $transient_name variable in set_transient

sprclldr opened the pull request:
I could only get the transient to store and retrieve correctly with its hashed name by declaring the $transient_name variable. Considering my previous issue, this could also be a PHP 5.5 issue.

joshuadavidnelson commented:
@sprclldr Thanks for the contribution!

The transient name isn't actually a class variable, though - it's generated on the function call via the private `transient_name` function and not stored. Odd that adding that line would work, but maybe I'm missing something...

Seems like it might be easier to modify that function to store/retrieve the transient name with the class `request_url` instead of using a url argument, especially since it's currently a private function and the request url is the only one every passed to it.

Perhaps the `transient_name` function should be modified to this:
```php
	/**
	 * Set the transient name.
	 * 
	 * @since 1.0.0
	 * 
	 * @param string $url The request url.
	 * 
	 * @return string $transient_name
	 */
	public function transient_name() {
		return $this->cache_prefix . md5( $this->request_url );
	}

```

And/or we could use this function as a fallback and provide a `transient_name` as a class argument.

What do you think?

sprclldr commented:
I ran a diff and realized I'd forgotten my edit on line 143. Just committed that for reference, which may explain why it's working. I'm an OOP noob, so reverse engineering isn't my strong suit. 

I'm testing your proposal now.

joshuadavidnelson commented:
Cool, sounds good. I'll admit that my OOP mastery is still in process, as well :wink:

I haven't touched this class in a few weeks and I'm away from my computer to run tests, but there's probably a better way to tackle it. Let me know if you figure one out, and I'll make a run at it later this evening or tomorrow.

Cheers!

sprclldr commented:
Okay, this last commit pinpoints and fixes the issue. I've removed the variable declaration and replaced the call to `$this->transient_name` in line 136. Maybe it wasn't being defined in that scope?

joshuadavidnelson commented:
@sprclldr Sweet! I'll take a closer look at everything this evening and test it. Appreciate the contribution!

joshuadavidnelson commented:
I was just looked at the updated PR and this makes more sense, I was confused by the use of `$this->transient_name` and failed to realize _I was using it_ when setting the transient. I believe the transient was working mostly be accident! You're updated PR fixes my misuse of the non-existent class variable.

I'll merge this and probably come back to the entire transient naming method at some point in the future. I like the idea of allowing the user to override the transient name via an argument, but this works for now.
